cpp/src/arrow/util/small_vector_test.cc

```diff
@@ -412,11 +412,6 @@ class TestSmallStaticVector : public ::testing::Test {
     IntVectorType<N> moved_moved_ints = std::move(moved_ints);
     ASSERT_EQ(moved_moved_ints.size(), 5);
     EXPECT_THAT(moved_moved_ints, ElementsAre(4, 5, 6, 7, 8));
-
-    // Move into itself
```

Comment:
The cpp standard makes it so that standard library types that are self assigned like this exist in a valid but unspecified state, so I think testing the contents of this after a self move does not rely upon standard behavior.

See also https://stackoverflow.com/questions/68274999/is-it-allowed-to-self-move-an-object-in-c